R/plotPValHist.R

```diff
@@ -25,110 +24,146 @@
 #' }
 #'
 #' @import ggplot2
-#' @importFrom dplyr filter
+#' @importFrom dplyr filter select
 #' @importFrom canvasXpress canvasXpress
 #'
 #' @export
 plotPvalHist <- function(P.Val,
                          plotType = "canvasXpress",
-                         facet = TRUE,
-                         binWidth = 0.02,
-                         alpha = 0.6,
-                         color = "dodgerblue3",
-                         fill = "dodgerblue3") {
+                         facet          = TRUE,
+                         binWidth       = 0.02,
+                         transparency   = 0.6,
+                         color    = "dodgerblue3") {
 
-    assertthat::assert_that(plotType %in% c("canvasXpress", "ggplot"),
+    plotType = tolower(plotType)
+
+    assertthat::assert_that(!missing(P.Val),
+                            class(P.Val)[[1]] %in% c("matrix","data.frame"),
+                            msg = "P.Val must be specified and must be of class matrix or dataframe.")
+
+    assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
                             msg = "Plot type must be either canvasXpress or ggplot.")
 
+    if (!assertthat::see_if(is.character(color),
+                            length(color) == 1)) {
+        warning("color must be a singular value of class character and must specify the name of the color or the rgb value. Assigning default value 'dodgerblue3'.")
+        color <- "dodgerblue3"
+    }
+
+    if (!assertthat::see_if(is.logical(facet),
+                            length(facet) == 1)) {
+        warning("facet must be a singular logical value. Assigning default value TRUE.")
+        facet <- TRUE
+    }
+
+    if (!assertthat::see_if(is.numeric(binWidth),
+                            length(binWidth) == 1,
+                            binWidth > 0 & binWidth <= 1)) {
```

Comment:
do not use bitwise **&**, Connie does not like it, you can use either normal and **&&** or just a comma here as all the conditions are **anded** here

Reply:
Updated.